### PR TITLE
Bugfix/json error handling fix

### PR DESCRIPTION
**Pull request by vladzz**
Add's a new exception which handles multiple downstream errors or returns as before if there is only one error.

### CODE REVIEW

File: src/main/scala/com/redbubble/util/http/JsonApiClient.scala [hunk 1]
```diff
@@ -9,16 +9,40 @@ import com.redbubble.util.metrics.StatsReceiver
...
+    val multiError = extractSeveralErrors(c)
+    val singleError = extractSingleError(c)
+
+    val singleErrorResult = singleError.map(s =>
+      Errors.downstreamError(new Exception(s), interaction(url, headers, None, response))
+    )
+
+    val someSortOfError = singleErrorResult match {
+      case Some(err) => err
+      case None => Errors.downstreamError(GenericMultipleErrors(s"Multiple errors returned downstream", None, multiError), interaction(url, headers, None, response))
```
Comment (TJC) [inline src/main/scala/com/redbubble/util/http/JsonApiClient.scala]: I would consider changing that `case None` to be
`case None if multiError.nonEmpty =>`
and then add below it something like
`case None => someKindOfDefaultError`

to catch the case where you've ended up here with neither of the single or multi errors in the json.

**Review by TJC — COMMENTED** [review]
Hope you don't mind me jumping in -- just had a minor suggestion for the error handling

**Comment by vladzz** [comment]
@TJC Updated to include that extra case. Is this okay now?

**Comment by kenvandenenden** [comment]
LGTM 👍

**Pull request merged**